src/descriptors.rs

```diff
@@ -333,12 +434,103 @@ impl PartialEq<descriptor::Descriptor<descriptor::DescriptorPublicKey>> for Inhe
     }
 }
 
+/// Information about a single spending path in the descriptor.
+#[derive(Debug, Eq, PartialEq, Clone, Ord, PartialOrd, Hash)]
+pub enum PathInfo {
+    Single(descriptor::DescriptorPublicKey),
+    Multi(usize, Vec<descriptor::DescriptorPublicKey>),
+}
+
+impl PathInfo {
+    /// Get the spending path info from a policy describing a single key or a multisig.
+    /// Will return None if the policy isn't a key or a multisig.
+    pub fn from_single_key_or_multisig(
+        policy: SemanticPolicy<descriptor::DescriptorPublicKey>,
+    ) -> Option<PathInfo> {
+        match policy {
+            SemanticPolicy::Key(key) => Some(PathInfo::Single(key)),
+            SemanticPolicy::Threshold(k, subs) => {
+                let keys: Option<Vec<descriptor::DescriptorPublicKey>> = subs
+                    .into_iter()
+                    .map(|sub| match sub {
+                        SemanticPolicy::Key(key) => Some(key),
+                        _ => None,
+                    })
+                    .collect();
+                Some(PathInfo::Multi(k, keys?))
+            }
+            _ => None,
+        }
+    }
+
+    /// Get the required number of keys for spending through this path, and the set of keys
+    /// that can be used to provide a signature for this path.
+    pub fn thresh_fingerprints(&self) -> (usize, HashSet<bip32::Fingerprint>) {
+        match self {
+            PathInfo::Single(key) => {
+                let mut fingerprints = HashSet::with_capacity(1);
+                fingerprints.insert(key_fingerprint(key).expect("Must be a multixpub."));
+                (1, fingerprints)
+            }
+            PathInfo::Multi(k, keys) => (
+                *k,
+                keys.iter()
+                    .map(|key| key_fingerprint(key).expect("Must be a multixpub."))
+                    .collect(),
+            ),
+        }
+    }
+}
+
+/// Information about the descriptor: how many keys are present in each path, what's the timelock
+/// of the recovery path, what's the threshold if there are multiple keys, etc..
+#[derive(Debug, Eq, PartialEq, Clone, Ord, PartialOrd, Hash)]
+pub struct LianaDescInfo {
+    primary_path: PathInfo,
+    recovery_path: (u16, PathInfo),
+}
+
+impl LianaDescInfo {
+    pub(crate) fn new(primary_path: PathInfo, recovery_path: (u16, PathInfo)) -> LianaDescInfo {
+        LianaDescInfo {
+            primary_path,
+            recovery_path,
+        }
+    }
+
+    pub fn primary_path(&self) -> &PathInfo {
+        &self.primary_path
+    }
+
+    /// Timelock and path info for the recovery path.
+    pub fn recovery_path(&self) -> (u16, &PathInfo) {
+        (self.recovery_path.0, &self.recovery_path.1)
+    }
+}
+
+/// Number of signatures present and required number of signatures for a given path.
+#[derive(Debug, Eq, PartialEq, Clone, Copy, Ord, PartialOrd, Hash)]
+pub struct PathSpendInfo {
+    pub sigs_count: usize,
```

Comment:
It is more useful for the GUI to provide the (fingerprint,derivation_path) of each signing key. So user can know which one is missing.